### ISSUE TITLE
Wizard component validation

### ISSUE THREAD

**Issue by rowdy4e** [laravel-filament/filament#4496]
### Package

filament/filament

### Package Version

v2.16.31

### Laravel Version

9.35.1

### PHP Version

8.1.11

### Problem description

Wizard component returns to the last step even though there are errors from validations in the previous steps.

### Expected behavior

The wizard should leave you at the step where the validation is performed(if there is some validation error) using the ENTER.

### Steps to reproduce

You can try it on the demo project https://demo.filamentphp.com/shop/orders/create (click on Street Address Input and hit enter)

1. Create page `php artisan make:filament-page TestPage`
2. Edit `resources/views/filament/pages/test-page.blade.php` to following code.
```
<x-filament::page>
    <form wire:submit.prevent="submit">
        {{ $this->form }}
    </form>
</x-filament::page>
```
3. Edit `app/Filament/Pages/TestPage.php`
```
...

class TestPage extends Page
{
    ....
    
    public function mount()
    {
        $this->form->fill();
    }

    public $state;

    protected function getFormStatePath(): ?string
    {
        return 'state';
    }

    protected function getFormSchema(): array
    {
        return [
            Wizard::make([
                Wizard\Step::make('Step #1')->schema([
                    TextInput::make('input_one')
                        ->required()
                ]),
                Wizard\Step::make('Step #2')->schema([
                    TextInput::make('input_two')
                        ->required()
                ]),
                Wizard\Step::make('Step #3')->schema([
                    TextInput::make('input_three')
                        ->required()
                ])
            ])->submitAction(new HtmlString('<button type="submit">Submit</submit>'))
        ];
    }

    public function submit()
    {
        $this->form->validate();

        dd('Submitted');
    }
}

```
4. Go to http://127.0.0.1:8000/admin/test-page
5. Click into first TextInput and hit Enter.

### Reproduction repository

https://github.com/rowdy4e/filament-wizard-bug

**Comment by zepfietje** [comment]
Reproducible on Chromium macOS (not on Safari).

**Comment by LukasJK** [comment]
Reproducible on Chrome Windows when there are 3 or more wizard steps. Works fine with just 2

**Comment by danharrin** [comment]
Fixed by #4916
